### PR TITLE
test: Add tests for golang and react with in memory files

### LINKED ISSUE #42: Add tests validating output

### PR DESCRIPTION
**Pull request by anghelflorinm**
## This PR

- adds tests for React and Golang

### Related Issues
https://github.com/open-feature/codegen/issues/42

Fixes #42 

### How to test
`go test ./...` from root folder

### CODE REVIEW

File: cmd/generate/generate_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+)
+
+func TestGenerateGoSucces(t *testing.T) {
```
Comment (beeme1mr) [inline cmd/generate/generate_test.go:15]:
```suggestion
func TestGenerateGoSuccess(t *testing.T) {
```

File: cmd/generate/generate_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	})
+
+	// Run command.
+	Root.Execute()
+
+	// Compare result.
+	compareOutput(t, testFileGo, memoryOutputPath, fs)
+}
+
+func TestGenerateReactSucces(t *testing.T) {
```
Comment (beeme1mr) [inline cmd/generate/generate_test.go:42]:
```suggestion
func TestGenerateReactSuccess(t *testing.T) {
```

File: internal/generate/generate.go [hunk 3]
```diff
@@ -28,10 +28,11 @@ func GenerateFile(funcs template.FuncMap, contents string, data types.TmplDataIn
 		return fmt.Errorf("error executing template: %v", err)
 	}
 	outputPath := data.BaseTmplDataInfo().OutputPath
-	if err := os.MkdirAll(filepath.Dir(outputPath), 0770); err != nil {
+	fs := filesystem.FileSystem()
+	if err := fs.MkdirAll(filepath.Dir(outputPath), 0770); err != nil {
```
Comment (beeme1mr) [inline internal/generate/generate.go:32]: Does this directory execute permission?

Comment (beeme1mr) [reply internal/generate/generate.go:32]: I think `0660` would be better.

Comment (anghelflorinm) [reply internal/generate/generate.go]: Updated

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**